### DISCUSSION TITLE
PDF Export issue

### DISCUSSION THREAD

**Discussion by shortsam1** [serenity-is/Serenity#5826] [Help] [answered]
I implemented the following code into my solution to export to PDF, but I get the following error which I cannot solve.

`            buttons.push(SparcOnlineForms.Common.PdfExportHelper.createToolButton({
                grid: this,
                onViewSubmit: () => this.onViewSubmit()
            }));`

![image](https://user-images.githubusercontent.com/47765946/123361602-b7d4be80-d5b2-11eb-8f43-02c3e03e00d2.png)
![image](https://user-images.githubusercontent.com/47765946/123361783-fcf8f080-d5b2-11eb-8949-3a29c6d738e5.png)

I checked and **/Scripts/jspdf.min.js** exists.

**Comment by VictorTomaili** [comment, marked as answer]
`Install-Package Serenity.Extensions -Version 5.0.43`

**Reply by shortsam1** [reply]
I have installed the package, cleaned and rebuilt the solution but still get the same error.  Am I missing something else?
![image](https://user-images.githubusercontent.com/47765946/124041205-c5be8f80-da49-11eb-8178-4a0bb469e13b.png)

Error:
![image](https://user-images.githubusercontent.com/47765946/124042163-9dd02b80-da4b-11eb-9a86-3a56078775a1.png)

File:
![image](https://user-images.githubusercontent.com/47765946/124042183-a6c0fd00-da4b-11eb-9f05-0c0aa9c6559f.png)

**Reply by VictorTomaili** [reply]
did you restore the assets with `dotnet tool update sergen` then `dotnet sergen restore`

**Reply by shortsam1** [reply]
I used the nuget manager within visual studio.  I removed the extensions and installed them again, cleaned and rebuilt the solution again and they have popluated into the following directory.

![image](https://user-images.githubusercontent.com/47765946/124196065-ff0d0300-db0e-11eb-8c33-f1632927ef14.png)

They are now working.

Thanks for your help @VictorTomaili.  I would not of got there without knowing about having to install the extensions.
